### PR TITLE
Add VPC attachment appliance mode support

### PR DESCRIPTION
**Pull request by lgallard**
## what
* Add VPC attachment appliance mode support

## why
* To enable traffic flow between  source and destination in the same Availability Zone for the VPC attachment for the lifetime of that flow.

## references
* https://docs.aws.amazon.com/vpc/latest/tgw/transit-gateway-appliance-scenario.html

### CODE REVIEW

**Comment by lgallard** [comment]
@nitrocode @Nuru we ran into a traffic performance issue  and realized the TF resource attribute was missing in the module.  Can you review it to merge it into your code? 

Thanks ins advance.

PS: We checked at our end and couldn't find any option to enable commits form the upstream maintainers


cc: @exequielrafaela @diego-ojeda-binbash

File: main.tf [hunk 1]
```diff
@@ -40,13 +40,14 @@ data "aws_vpc" "default" {
...
-  vpc_id             = each.value["vpc_id"]
-  subnet_ids         = each.value["subnet_ids"]
-  dns_support        = var.vpc_attachment_dns_support
-  ipv6_support       = var.vpc_attachment_ipv6_support
-  tags               = module.this.tags
+  for_each               = module.this.enabled && var.create_transit_gateway_vpc_attachment && var.config != null ? var.config : {}
+  transit_gateway_id     = local.transit_gateway_id
+  vpc_id                 = each.value["vpc_id"]
+  subnet_ids             = each.value["subnet_ids"]
+  appliance_mode_support = var.appliance_mode_support
```
Comment (nitrocode) [inline main.tf:47]:
```suggestion
  appliance_mode_support = var.vpc_attachment_appliance_mode_support
```

**Review by nitrocode — DISMISSED** [review]
@lgallard Thanks for the contribution. Please see comment.

File: main.tf (hunk 1 above)
Comment (lgallard) [reply main.tf]: @nitrocode nice catch!  Done!

**Review by nitrocode — APPROVED** [review]

**Comment by exequielrafaela** [comment]
Great job @lgallard 💪🏼 @nitrocode Thanks! 🙏🏼 🙌🏼

**Pull request merged**